webapp/app/components/test-item/component.js

```diff
@@ -25,5 +26,16 @@ export default Component.extend({
       returned += "/errors";
     }
     return returned;
-  }.property("test")
+  }.property("test"),
+
+  actions: {
+    toggle_starred: function() {
+      self = this;
+      return this.get("api")
+        .call("toggle_starred", { test_id: this.get("test.id") })
+        .then(async function() {
+            await self.get('test').reload();
```

Comment:
IIRC you can just return the promise and it will work exactly the same as the async/await pair here